src/silx/gui/dialog/ColormapDialog.py

```diff
@@ -974,6 +977,23 @@ def __init__(self, parent=None, title="Colormap Dialog"):
         self._autoButtons = _AutoScaleButton(self)
         self._autoButtons.autoRangeChanged.connect(self._autoRangeButtonsUpdated)
 
+        # used percentile
+        self._centralPercentileWidget = SliderWithSpinBox(self)
```

Comment:
It sounds that converting `SliderWithSpinBox` to a specific widget, e.g., `ColormapPercentileWidget` would allow to move the code handling the percentile from `ColormapDialog` to this widget:
All the init code for sure, but also the conversion code to/from 1 "saturation" value to a percentile range.